src/pivot-table/hooks/use-data-model.ts

```diff
@@ -82,43 +84,45 @@ export default function useDataModel({
   );
 
   const applyColumnWidth = useCallback<ApplyColumnWidth>(
-    (newColumnWidth, { dimensionInfoIndex, isLeftColumn, x = 0 }) => {
-      const { qMeasureInfo, qDimensionInfo } = layoutService.layout.qHyperCube;
-      const isPseudoDimension = dimensionInfoIndex === PSEUDO_DIMENSION_INDEX;
-      let indexes: number[];
-
-      if (isPseudoDimension) {
-        indexes = isLeftColumn
-          ? [...Array(qMeasureInfo.length).keys()] // apply column width to all measures, since they are in the same column
-          : [layoutService.getMeasureInfoIndexFromCellIndex(x)];
-      } else {
-        // cell indexes don't correspond to dimension indexes, so we need to compensate for potential prior pseudo dim (and left side dims)
-        indexes = [dimensionInfoIndex];
+    (newColumnWidth, { dimensionInfoIndex, isLeftColumn, x = 0, columnWidthLocation }) => {
+      const { qMeasureInfo, qDimensionInfo, topHeadersColumnWidth } = layoutService.layout.qHyperCube;
+      let paths: { qPath: string; oldValue?: ColumnWidth }[] = [];
+
+      switch (columnWidthLocation) {
+        case ColumnWidthLocation.Dimension:
+          paths = [
+            {
+              qPath: `${Q_PATH}/qDimensions/${dimensionInfoIndex}/qDef/columnWidth`,
+              oldValue: qDimensionInfo[dimensionInfoIndex].columnWidth,
+            },
+          ];
+          break;
+        case ColumnWidthLocation.Measures:
+          if (isLeftColumn) {
+            paths = [...Array(qMeasureInfo.length).keys()].map((idx) => ({
```

Comment:
I would think this will give the same output

```suggestion
            paths = qMeasureInfo.map((info, idx) => ({
```

Reply:
Yes